### ISSUE TITLE
Proj0028: Conditional TargetFrameworks should not be reported on

### ISSUE THREAD

**Issue by Corniel** [dotnet-project-file-analyzers/dotnet-project-file-analyzers#257]
As discussed with @sungam3r at [destructurama/attributed](https://github.com/destructurama/attributed/pull/145):

``` XML
<PropertyGroup>
    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">net9.0;net462</TargetFrameworks>
    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">net9.0</TargetFrameworks>
</PropertyGroup>
```

Is easier to read then:

``` XML
<PropertyGroup Condition="'$(OS)' == 'Windows_NT'">
    <TargetFrameworks>net8.0;netstandard</TargetFrameworks>
<PropertyGroup>
<PropertyGroup Condition="'$(OS)' != 'Windows_NT'">
    <TargetFramework>net8.0</TargetFramework>
</PropertyGroup>
```

or  

``` XML
<Choose>
  <When Condition="'$(OS)' == 'Windows_NT'">
    <PropertyGroup>
        <TargetFrameworks>net8.0;netstandard</TargetFrameworks>
    </PropertyGroup>
  </When>
  <Otherwise>
    <PropertyGroup>
        <TargetFramework Condition="'$(OS)'  != 'Windows_NT'">net8.0</TargetFramework>
    </PropertyGroup>
  </Otherwise>
</Choose>
```

Therefor, it might be okay to allow `<PropertyGroup>`'s to have children with conditions if - and only if - **all** children are conditional.

**Comment by sungam3r** [comment]
> if and only if

Are you sure?

**Comment by Corniel** [comment]
> > if and only if
> 
> Are you sure?

The reasoning about this rule is that conditionals should be easy to spot, and should not obfuscate the configuration they are setting. By allowing conditionals at level 2 under the premise that all items in the `<PropertyGroup>`  should be conditional, I think the intention of the rule stays valid. You do not agree?

**Comment by sungam3r** [comment]
Is that code compliant?
```
<PropertyGroup>
    <TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0;net8.0;net9.0</TargetFrameworks>
    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">$(TargetFrameworks);net461;net462;net48</TargetFrameworks>
</PropertyGroup>
```

It's a common pattern to append tfms on case of Windows to test code on many pkatforms.
